sorting/count_inversions.cpp

```diff
@@ -0,0 +1,191 @@
+/**
+ * @file
+ * @brief Counting Inversions using [Merge Sort](https://en.wikipedia.org/wiki/Merge_sort)
+ *
+ * Program to count the number of inversions in an array
+ * using merge-sort.
+ * 
+ * The count of inversions help to determine how close the array
+ * is to being sorted in ASCENDING order.
+ * 
+ * two elements a[i] and a[j] form an inversion if a[i] > a[j] and i < j
+ * 
+ * Time Complexity --> O(n.log n)
+ * Space Complexity --> O(n) ; additional arrat temp[1..n]
+ * Algorithm:
+ *   1. The idea is similar to merge sort, divide the array into two equal or almost 
+ *      equal halves in each step until the base case is reached.
+ *   2. Create a function merge that counts the number of inversions when two halves of
+ *      the array are merged, create two indices i and j, i is the index for first half 
+ *      and j is an index of the second half. if a[i] is greater than a[j], then there are (mid – i)  
+ *      inversions, Because left and right subarrays are sorted, so all the remaining elements 
+ *      in left-subarray (a[i+1], a[i+2] … a[mid]) will be greater than a[j].
+ *   3. Create a recursive function to divide the array into halves and find the answer by summing 
+ *      the number of inversions is the first half, number of inversion in the second half and 
+ *      the number of inversions by merging the two.
+ *   4. The base case of recursion is when there is only one element in the given half.
+ *   5. Print the answer
+ * 
+ */
+#include <iostream>
+#include <vector>
+
+int sorting::inversion::mergeSort(int* arr, int* temp, int left, int right);
+int sorting::inversion::merge(int* arr, int* temp, int left, int mid, int right);
+int sorting::inversion::countInversion(int* arr, const int size);
+void sorting::inversion::show(int* arr, const int size);
```

Comment:
Not needed to add the namespace names at the beginning if these functions are inside the namespace.